src/Util/MakerFileLinkFormatter.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/*
+ * This file is part of the Symfony MakerBundle package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\MakerBundle\Util;
+
+use Symfony\Component\HttpKernel\Debug\FileLinkFormatter;
+
+final class MakerFileLinkFormatter
+{
+    private $fileLinkFormatter;
+    private $fileLinkFormat;
+
+    public function __construct(FileLinkFormatter $fileLinkFormatter = null, string $fileLinkFormat = null)
+    {
+        // Since nullable types are not available in 7.0; can be removed when php >= 7.1 required
+        if (\func_num_args() < 2) {
+            throw new \LogicException('$fileLinkFormatter and $fileLinkFormat arguments are required');
+        }
+
+        $this->fileLinkFormatter = $fileLinkFormatter;
+        $this->fileLinkFormat = $fileLinkFormat;
+    }
+
+    public function makeLinkedPath(string $absolutePath, string $relativePath): string
+    {
+        if ($this->fileLinkFormat) {
+            if ($this->fileLinkFormatter) {
+                return $this->createLink(
+                    $relativePath,
+                    $this->fileLinkFormatter->format($absolutePath, 1)
+                );
+            }
+
+            return $this->createLink(
+                $relativePath,
+                strtr($this->fileLinkFormat, ['%f' => $absolutePath, '%l' => 1])
+            );
+        }
+
+        return $relativePath;
+    }
```

Comment:
I think we can reverse a bunch of this logic to exit earlier and avoid nesting. I also think we can probably do NOTHING if there is no `fileLinkFormatter` - it would be a super edge case to have a `fileLinkFormat` but *not* a `fileLinkFormatter`:

```php
if (null === $this->fileLinkFormatter || null === $this->fileLinkFormat) {
    return $relativePath;
}

return $this->createLink(
     $relativePath,
     $this->fileLinkFormatter->format($absolutePath, 1)
);
```